client/src/components/payment/payment-modal.vue

```diff
@@ -103,6 +107,11 @@ export default class PaymentModal extends Mixins(
 
   /** Clears store property to hide this modal. */
   async hideModal () {
+    // stop timer because we don't need it any more
+    const { stopTimer } = this.$props
+    if (typeof stopTimer === 'function') {
+      stopTimer()
+    }
```

Comment:
I still don't quite follow these styles of conditions. Why do we check it in this manner if it's defined above as a function?

Reply:
In case it's not passed in as a prop, I think.
I haven't verified whether this working properly (existing code).

Reply:
ok cool, thanks!